$OpenBSD: patch-configure,v 1.9 2019/12/04 10:36:34 sthen Exp $

The test for __builtin_bswap64 must fail if the linker can't find the
symbol.  We need this for base-gcc on little endian, like mips64el.

Fix relocation errors on (at least) sparc64.

We don't want extra debug flags in regular builds.

Index: configure
--- configure.orig
+++ configure
@@ -6062,7 +6062,7 @@ uint64_t y = __builtin_bswap64(x);
   return 0;
 }
 _ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
+if ac_fn_c_try_link "$LINENO"; then :
   nettle_cv_c_builtin_bswap64=yes
 else
   nettle_cv_c_builtin_bswap64=no
@@ -6720,6 +6720,7 @@ else
 	bsdi4.*)	CCPIC="-fPIC" ;;
 	bsdi*)		CCPIC="" ;;
 	darwin*)	CCPIC="-fPIC" ;;
+	openbsd*)	CCPIC="-fPIC" ;;
 	# Could also use -fpic, depending on the number of symbol references
 	solaris*)	CCPIC="-fPIC" ;;
 	cygwin*)	CCPIC="" ;;
@@ -7407,7 +7408,7 @@ if test x$GCC = xyes ; then
   if $CC --version | grep '^2\.96$' 1>/dev/null 2>&1; then
     true
   else
-    CFLAGS="$CFLAGS -ggdb3"
+    CFLAGS="$CFLAGS"
   fi
   # FIXME: It would be better to actually test if this option works and/or is needed.
   # Or perhaps use -funsigned-char.
